plugin-adscc/src/main/java/org/apache/ranger/services/adscc/AdsccRestService.java

```diff
@@ -0,0 +1,69 @@
+package org.apache.ranger.services.adscc;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.log4j.Logger;
+import org.apache.ranger.plugin.util.PasswordUtils;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Optional;
+
+public class AdsccRestService {
+    private static final Logger logger = Logger.getLogger(AdsccRestService.class);
+    private final HttpClient httpClient;
+
+    public AdsccRestService(final HttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    public Optional<JsonElement> execute(final String url,
+                                         final String username,
+                                         final String password) {
+        try {
+            return Optional.ofNullable(JsonParser.parseString(IOUtils.toString(getResponse(url, username, password).getEntity().getContent(), StandardCharsets.UTF_8)));
```

Comment:
I suggest moving nested objects to separate variables to increase readability

Reply:
Fixed.